google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITTransactionManagerTest.java

```diff
@@ -212,4 +214,31 @@ public void abortAndRetry() throws InterruptedException {
       manager2.close();
     }
   }
+
+  @SuppressWarnings("resource")
+  @Test
+  public void testTransactionManagerReturnsCommitStats() throws InterruptedException {
+    assumeFalse("Emulator does not return commit statistics", isUsingEmulator());
+    try (TransactionManager manager = client.transactionManager(Options.commitStats())) {
+      TransactionContext transaction = manager.begin();
+      while (true) {
```

Comment:
This one scares me since it looks like an infinite loop Could you add a counter with a maximum number of retries before failure?

Reply:
This is the recommended way to use `TransactionManager` according to the [documentation](https://github.com/googleapis/java-spanner/blob/5af36739532037360dfd504a4a0988562550526c/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java#L324). It is also the way it is already used in multiple other test cases. I would rather either:
1. Keep this and all other instances as they are.
2. Change this and other instances + the documentation in a separate PR.